src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,46 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters");
```

Comment:
The check for the login length should be removed, as it is not described in the task requirements. The task specifies only the password length constraint, not the login.